synapse/event_auth.py

```diff
@@ -29,18 +29,19 @@
     EventFormatVersions,
     RoomVersion,
 )
+from synapse.events import EventBase
 from synapse.types import UserID, get_domain_from_id
 
 logger = logging.getLogger(__name__)
 
 
 def check(
     room_version_obj: RoomVersion,
-    event,
-    auth_events,
-    do_sig_check=True,
-    do_size_check=True,
-):
+    event: EventBase,
+    auth_events: Dict[Tuple[str, str], EventBase],
```

Comment:
this can also be spelt:

```suggestion
    auth_events: StateMap[EventBase],
```

Likewise below. Up to you on whether to go and replace them all.